### DISCUSSION TITLE
Proper package publishing

### DISCUSSION THREAD

**Discussion by ComradePashka** [verze-app/solana-php-sdk#44] [General]
Hello!

I decided to use this package in my project, so I followed instruction but seems that `tightenco/solana-php-sdk` in fact is abandoned and there is no `verze-app/solana-php-sdk` at packagist.org

This a bit confusing. Does it mean that with `composer require tightenco/solana-php-sdk` we actually never use **THIS** repo?

Should it be fixed? Should `verze-app/solana-php-sdk` be registered package at packagist.org and be suggested replacement for the former original package? Or at least should installation instruction be fixed to target this repo instead of former one?

Thank you in advance,
cheers!

P.S. idk if package name is already somehow automatically target this repo, so post here instead of creating issue.

**Comment by ComradePashka** [comment]
UPD

So, I made a little research and find out that installation instruction will work properly only when repo section is added to `composer.json`:

```
    "repositories": [
        {
            "type":"package",
            "package": {
                "name": "tightenco/solana-php-sdk",
                "version":"master",
                "source": {
                    "url": "https://github.com/verze-app/solana-php-sdk.git",
                    "type": "git",
                    "reference":"main"
                }
            }
        }
    ],
```

without that it will use original package (due to old namespace `tightenco`), which is outdated significantly comparing to this repo.

**Comment by ComradePashka** [comment]
UPD2

but anyway with this method there's still some issues with namespaces/classnames -- it cannot find any class from Tighten\SolanaPhpSdk in runtime :-(

**Comment by ComradePashka** [comment]
UPD3

just checked other repo and seems when targeting exact version in composer it is working: "tightenco/solana-php-sdk": "^0.3.2"
however maybe worth to mention about it in documentation?

**Comment by ComradePashka** [comment]
UPD4!

in previous config there's some autoload issues, so even tho it's installed by composer, then package namespaces and classes not available in the project, so seems proper config is to have repo in composer like this:
```
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/verze-app/solana-php-sdk.git"
        }
    ],
```
and target latest version:
```
"tightenco/solana-php-sdk": "dev-main"
```

**Comment by neverything** [comment]
Hi @ComradePashka, thanks for the updates, yes we are still working on migrating to the new namespace and publishing a new release on packagist.
